grpclog/grpclog.go

```diff
@@ -218,3 +226,22 @@ func (l *GrpcLog) Marshal() ([]byte, error) {
 	}
 	return bytes, nil
 }
+
+type hasURN interface {
+	GetUrn() string
+}
+
+// FindByUrn finds the first entry with the given resource URN, or nil if none is found.
+func FindByUrn[TRequest resourceRequest, TResponse resourceResponse](entries []TypedEntry[TRequest, TResponse],
+	urn string) *TypedEntry[TRequest, TResponse] {
+	// nolint:copylocks
+	for _, e := range entries {
+		var eI any = &e.Request
+		if hasUrn, ok := eI.(hasURN); ok {
+			if hasUrn.GetUrn() == urn {
+				return &e
+			}
+		}
```

Comment:
Use a type switch to test for interfaces:

```suggestion
        switch r := e.Request.(type) {
        case hasURN:
            if r.GetURN() == urn {
                return &e
            }
        }
```